src/services/expenses.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const { client } = require('../utils/db');
+
+function normalize({
+  id,
+  user_id: userId,
+  spent_at: spentAt,
+  title,
+  amount,
+  category,
+  note,
+}) {
+  return {
+    id,
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+}
+
+async function getMany(userId, category, from, to) {
+  const queryArray = [];
+  const parametersArray = [...arguments].filter(parameter => parameter).length
+    ? [...arguments]
+    : [];
+
+  queryArray.push(userId ? 'user_id=$1' : '');
+  queryArray.push(category ? 'category=$2' : '');
+  queryArray.push(from ? 'spent_at>$3' : '');
+  queryArray.push(to ? 'spent_at<$4' : '');
+
+  const queryString = queryArray
+    .filter(query => query)
+    .join(' AND ')
+      || 'id>0';
+
+  const result = await client.query(`
+    SELECT *
+    FROM expenses
+    WHERE ${queryString}
+    ORDER BY id
+    `, [...parametersArray]
+  );
+
+  return result.rows;
+}
+
+async function getOne(expenseId) {
+  const result = await client.query(`
+    SELECT *
+    FROM expenses
+    WHERE id=$1
+    `, [expenseId]
+  );
+
+  return result.rows[0] || null;
+}
+
+async function create(...values) {
+  const valuesString = values
+    .map((arg, i) => `$${i + 1}`)
+    .join(',');
+
+  await client.query(`
+    INSERT INTO expenses(user_id, spent_at, title, amount, category, note)
+    VALUES (${valuesString})
+    `, [...values]
+  );
+
+  const result = await client.query(`
+    SELECT *
+    FROM expenses
+    ORDER BY id DESC
+    LIMIT 1
+  `);
+
+  return result.rows[0];
```

Comment:
```suggestion
  const result = await client.query(`
    INSERT INTO expenses(user_id, spent_at, title, amount, category, note)
    VALUES (${valuesString})
    `, [...values]
  );

  return result.rows[0];
```
Insert query returns created expense, so you can get rid of the redundant query.

Reply:
And here the same))